Vostok.Metrics.System/Host/NetworkUtilizationCollector_Linux.cs

```diff
@@ -83,35 +135,109 @@ IEnumerable<NetworkUsage> FilterDisabledInterfaces(IEnumerable<NetworkUsage> int
                         long.TryParse(parts[1], out var receivedBytes) &&
                         long.TryParse(parts[9], out var sentBytes))
                     {
-                        networkInterfacesUsage.Add(
-                            new NetworkUsage
-                            {
-                                InterfaceName = parts[0].TrimEnd(':'),
-                                ReceivedBytes = receivedBytes,
-                                SentBytes = sentBytes
-                            });
+                        var interfaceName = parts[0].TrimEnd(':');
+                        networkInterfacesUsage[interfaceName] = CreateNetworkUsage(interfaceName, receivedBytes, sentBytes);
                     }
                 }
 
                 // NOTE: See https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-class-net for details.
                 // NOTE: This value equals -1 if interface is disabled, so we will filter this values out later.
-                foreach (var networkUsage in networkInterfacesUsage)
-                    networkUsage.NetworkMaxMBitsPerSecond = int.Parse(File.ReadAllText($"/sys/class/net/{networkUsage.InterfaceName}/speed"));
+                foreach (var networkUsage in networkInterfacesUsage.Values)
+                {
+                    if (TryReadSpeed(networkUsage.InterfaceName, out var result) && int.TryParse(result, out var speed))
+                        networkUsage.NetworkMaxMBitsPerSecond = speed;
+                    else
+                    {
+                        // NOTE: We don't need zero values. Mark this interface disabled for now. It's speed may be calculated later if it is a teaming interface.
+                        networkUsage.NetworkMaxMBitsPerSecond = -1;
+                    }
+                }
+
+                // NOTE: We don't check if interface speed was already calculated because we need ChildInterfaces property in any case.
+                var teamingInterfaces = networkInterfacesUsage.Values.OfType<TeamingNetworkUsage>();
+
+                foreach (var teamingInterface in teamingInterfaces)
+                    FillTeamingInfo(networkInterfacesUsage, teamingInterface);
```

Comment:
An exception from a single teaming interface is going to ruin the rest of the method. Shouldn't we just skip the failed one?